tools/docker-copy.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+INPUTS=("${@}")
+TARGET_ARCH=${TARGET_ARCH:-amd64}
+DOCKER_WORKING_DIR=${INPUTS[${#INPUTS[@]}-1]}
+FILES=${INPUTS[@]:0:${#INPUTS[@]}-1}
+
+function may_copy_into_arch_named_sub_dir() {
+  FILE=${1}
+  COPY_ARCH_RELATED=${COPY_ARCH_RELATED:-1}
+
+  FILE_INFO=$(file "${FILE}")
+  # when file is an executable, like binary,
+  # will put an arch named sub dir
+  # like
+  #   arm64/
+  #   amd64/
+  if [[ ${FILE_INFO} == *"executable"* ]]; then
+    chmod 755 "${FILE}"
+
+    case ${FILE_INFO} in
+      *x86-64*)
+        mkdir -p "${DOCKER_WORKING_DIR}/amd64/" && cp -rp "${FILE}" "${DOCKER_WORKING_DIR}/amd64/"
+        ;;
+      *aarch64*)
+        mkdir -p "${DOCKER_WORKING_DIR}/arm64/" && cp -rp "${FILE}" "${DOCKER_WORKING_DIR}/arm64/"
+        ;;
+        *)
+        cp -rp "${FILE}" "${DOCKER_WORKING_DIR}"
+        ;;
+    esac
+
+
+    if [[ ${COPY_ARCH_RELATED} == 1 ]]; then
+      # if other arch files exists, should copy too.
+      for ARCH in "amd64" "arm64"; do
+        # like file `out/linux_amd64/pilot-discovery`
+        # should check  `out/linux_arm64/pilot-discovery` exists then do copy
+
+        FILE_ARCH_RELATED=${FILE/linux_${TARGET_ARCH}/linux_${ARCH}}
+
+        if [[ ${FILE_ARCH_RELATED} != "${FILE}" && -f ${FILE_ARCH_RELATED} ]]; then
+          COPY_ARCH_RELATED=0 may_copy_into_arch_named_sub_dir "${FILE_ARCH_RELATED}"
+        fi
+      done
+    fi
+
```

Comment:
addtional copy for other archs, only when matched arch binary exists.

in this why, avoid to change Makefile too much, and now support single arch image builds